docs/Host_Validation_Architecture.md

```diff
@@ -0,0 +1,25 @@
+# Host Validation and Testing
+
+__Date:__  7/6/2017
+
+__Objective:__ To determine the best strategy to validate hosts have joined the Kubernetes cluster successfully and are ready to schedule workloads
+
+## Problem Statement
+The Promenade project is a simple and efficient tool for orchestrating the deployment of self-hosted Kubernetes clusters.  However, Promenade currently lacks the ability to validate that clusters have been deployed successfully and are actually ready to schedule workloads.   I would like to propose a series of tests and validations that can be included with Promenade which will ensure the Kubernetes clusters are working as intended at the time of deployment, and can be executed at any point in time  after the cluster has been deployed to ensure it is continuing to function as intended for Day 2 operations. This tool should be able to identify potential problems with the Kubernetes cluster and provide the user with a comprehensive debugging output which highlights the identified issues.
+
+## Identified Tests
+I have identified the following tests which I believe to be a good base-line for basic Kubernetes smoke testing at the time of cluster deployment and during cluster life-cycle management:
+
+1. __Generate SHA256 hashes of all Promenade Generated configuration files and certificates:__ This process will run during the initial deployment of all the Kubernetes hosts in the cluster. Promenade will write a file to `/var/log/promenade/promenade_file_hash.log` with the path to the file as well as the SHA256 hash of that file when it was originally generated. Subsequent runs of the host validation script will re-calculate the SHA256 hashes of the files as they are presently, and compare them with the values written in the log file at the time of deployment. If the file hashes differ, the host validation script will log a failure and provide the user with a list of files that have potentially been modified.
```

Comment:
@v1k0d3n your point is spot on in regard to the right algo for the job, but unfortunatly most code-analysis tools flag MD5 as insecure (le sigh) - therefore the extra couple of cyles spent hashing a config file seems inconsquential to the number of emails later. That said I'm still not sure what this method achives over running a `find` command - esp as looking at the WIP current code proposed you would be able to bypass this mechanism by simply deleteing/updating the  `/var/log/promenade/promenade_file_hash.log` file (and when you think about file perms, preventing this senario becomes difficult - as presumably the k8s configs should only be writeable by root ). So I think it needs to be specified at the outset if this is a security feature, or a sanity check - and if its the latter why it employs sich a complex mechanism for performing this this task.